Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs

```diff
@@ -150,72 +224,48 @@ private void UpdateSettings(BuildTarget currentBuildTarget)
-                    "OK", "Cancel");
+                EditorPrefsUtility.SetEditorPref(Names[ProjectSetting.UseInputManagerAxes], useToolkitAxes);
 
-                if (userPermission)
+                // Grabs the actual asset file into a SerializedObject, so we can iterate through it and edit it.
+                inputManagerAsset = new SerializedObject(AssetDatabase.LoadAssetAtPath("ProjectSettings/InputManager.asset", typeof(UnityEngine.Object)));
+
+                if (useToolkitAxes)
                 {
-                    using (var webRequest = UnityWebRequest.Get(InputManagerAssetURL))
+                    foreach (InputManagerAxis axis in newInputAxes)
                     {
-#if UNITY_2017_2_OR_NEWER
-                        webRequest.SendWebRequest();
-#else
-                        webRequest.Send();
-#endif
-
-                        while (!webRequest.isDone)
+                        if (!DoesAxisNameExist(axis.Name))
                         {
-                            if (webRequest.downloadProgress != -1)
-                            {
-                                EditorUtility.DisplayProgressBar("Downloading InputManager.asset from GitHub", "Progress...", webRequest.downloadProgress);
-                            }
+                            AddAxis(axis);
                         }
+                    }
 
-                        EditorUtility.ClearProgressBar();
-
-#if UNITY_2017_1_OR_NEWER
-                        if (webRequest.isNetworkError || webRequest.isHttpError)
-#else
-                            if (webRequest.isError)
-#endif
+                }
+                else
+                {
+                    foreach (InputManagerAxis axis in newInputAxes)
+                    {
+                        if (DoesAxisNameExist(axis.Name))
                         {
-                            Debug.LogError("Network Error: " + webRequest.error);
-                            userPermission = false;
+                            RemoveAxis(axis.Name);
                         }
-                        else
+                    }
+
+                    foreach (InputManagerAxis axis in obsoleteInputAxes)
+                    {
+                        if (DoesAxisNameExist(axis.Name))
                         {
-                            File.Copy(inputManagerPath, inputManagerPath + ".old", true);
-                            File.WriteAllText(inputManagerPath, webRequest.downloadHandler.text);
+                            RemoveAxis(axis.Name);
                         }
                     }
                 }
-            }
 
-            if (!userPermission)
-            {
-                Values[ProjectSetting.XboxControllerSupport] = false;
-                if (File.Exists(inputManagerPath + ".old"))
-                {
-                    File.Copy(inputManagerPath + ".old", inputManagerPath, true);
```

Comment:
I think we should probably make a backup copy anyway in case users want to revert their changes.

Reply:
If the box is unchecked, this script goes through and deletes all the Toolkit axes. Is that sufficient for reverting to the previous state?